gen2-efficentnet-classification/main.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+import argparse
+from pathlib import Path
+import sys
+import cv2
+import depthai as dai
+import numpy as np
+from classes import class_names
+
+# Get Argument First
+parser = argparse.ArgumentParser()
+parser.add_argument('-nd', '--no-debug', action="store_true", help="Prevent debug output")
+parser.add_argument('-cam', '--camera', action="store_true", help="Use DepthAI 4K RGB camera for inference (conflicts with -vid)")
+parser.add_argument('-vid', '--video', type=str, help="Path to video file to be used for inference (conflicts with -cam)")
+args = parser.parse_args()
+
+
+
+# NOTE: video must be of size 224 x 224. We will resize this on the
+# host, but you could also use ImageManip node to do it on device
+
+# Link video in with the detection network
+    
+if not args.camera and not args.video:
+    raise RuntimeError("No source selected. Please use either \"-cam\" to use RGB camera as a source or \"-vid <path>\" to run on video")
+
+debug = not args.no_debug
+camera = not args.video
+class_names = class_names()
+
+
+# Start defining a pipeline
+pipeline = dai.Pipeline()
+
+# NeuralNetwork
+print("Creating Neural Network...")
+detection_nn = pipeline.createNeuralNetwork()
+detection_nn.setBlobPath(str(Path("models/efficientnet-b0_openvino_2021.3_6shave.blob").resolve().absolute()))
+
+if camera:
+    print("Creating Color Camera...")
+    cam_rgb = pipeline.createColorCamera()
+    cam_rgb.setPreviewSize(224,224)
+    cam_rgb.setInterleaved(False)
+    cam_rgb.setResolution(dai.ColorCameraProperties.SensorResolution.THE_1080_P)
+    cam_rgb.setBoardSocket(dai.CameraBoardSocket.RGB)
+    cam_rgb.setColorOrder(dai.ColorCameraProperties.ColorOrder.RGB)
+
+    cam_xout = pipeline.createXLinkOut()
+    cam_xout.setStreamName("rgb")
+    cam_rgb.preview.link(cam_xout.input)
+    cam_rgb.preview.link(detection_nn.input)
+else:
+    face_in = pipeline.createXLinkIn()
+    face_in.setStreamName("in_nn")
+    face_in.out.link(detection_nn.input)
+
+# Create outputs
+xout_nn = pipeline.createXLinkOut()
+xout_nn.setStreamName("nn")
+detection_nn.out.link(xout_nn.input)
+
+frame = None
+bboxes = []
+
+
+def to_tensor_result(packet):
+    return {
+        tensor.name: np.array(packet.getLayerFp16(tensor.name)).reshape(tensor.dims)
+        for tensor in packet.getRaw().tensors
+    }
+
+
+def softmax(x):
+    """Compute softmax values for each sets of scores in x."""
+    e_x = np.exp(x - np.max(x))
+    return e_x / e_x.sum(axis=0)
+
+
+def to_planar(arr: np.ndarray, shape: tuple) -> np.ndarray:
+        resized = cv2.resize(arr, shape)
+        return resized.transpose(2, 0, 1)
+
+
+# Pipeline defined, now the device is assigned and pipeline is started
+with dai.Device(pipeline) as device:
+    device.startPipeline()
```

Comment:
you can remove this line
```
DeprecationWarning: Device(pipeline) starts the pipeline automatically. Use Device() and startPipeline(pipeline) otherwise
```

Reply:
Sure! I have changed the script as per suggestion :)